www/js/lib/zstddec_wrapper.js

```diff
@@ -0,0 +1,263 @@
+﻿/**
+ * zstddec_wrapper.js: Javascript wrapper around compiled zstd decompressor.
+ *
+ * Copyright 2020 Jaifroid, Mossroy and contributors
+ * License GPL v3:
+ *
+ * This file is part of Kiwix.
+ *
+ * Kiwix is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Kiwix is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Kiwix (file LICENSE-GPLv3.txt).  If not, see <http://www.gnu.org/licenses/>
+ */
+'use strict';
+define(['q', 'zstddec'], function(Q) {
+    // DEV: zstddec.js has been compiled with `-s EXPORT_NAME="ZD" -s MODULARIZE=1` to avoid a clash with xzdec which uses "Module" as its exported object
+    // Note that we include zstddec above in requireJS definition, but we cannot change the name in the function list
+    // There is no longer any need to load it in index.html
+    // For explanation of loading method below to avoid conflicts, see https://github.com/emscripten-core/emscripten/blob/master/src/settings.js
+    
+    /**
+     * @typedef EMSInstanceExt An object type representing an Emscripten instance with extended properties
+     * @property {Integer} _decHandle The decoder stream context object in asm memory (to be re-used for each decoder operation)
+     * @property {Object} _inBuffer A JS copy of the inBuffer structure to be set in asm memory (malloc)
+     * @property {Object} _outBuffer A JS copy of the outBuffer structure to be set in asm memory (malloc)
+     * @property {Integer} _chunkSize The number of compressed bytes to feed to the decompressor in any one read loop
+     */
+
+    /**
+     * The ZSTD Decoder instance
+     * @type EMSInstanceExt
+     */
```

Comment:
@mossroy This is a typedef I came up with. `{EMSInstanceExt}` stands for Emscripten Instance Extension.
IIt seems to work in the IDE: when I hover over `zd` anywhere in the code, its type pops up.
Thanks for the idea.

Reply:
NB it is commented above the visible code block here.